articles/golang-advanced-programming-concepts/index.md

```diff
@@ -0,0 +1,196 @@
+
+The syntax for creating a struct is as follows:
+
+```go
+type <name of the structure> struct {
+    field1 <data type>
+    field2 <data type>
+}
+```
+
+Example:
+
+```go
+package main
+
+import "fmt"
+
+type Student struct {
+    name string
+    age int
+    score float64
+}
+
+func main() {
+    var s1 = Student {
+        "Adith", 1, 99.5,
+    }
+    fmt.Println(s1)
+    fmt.Println(s1.name)
+    fmt.Println(s1.age)
+    fmt.Println(s1.score)
+
+    var s2 = Student {}
+    s2.name = "Bharadwaj"
+    fmt.Println(s2.name)
+}
+```
+
+In the above example, we create a struct called "Student" with three fields: `name`, `age`, and `score` of types `string`, `int`, and `float64`, respectively. In the main function, we create a variable called `s1` of type `Student` and initialize it with values that correspond to the `name`, `age`, and `score`. The curly braces({}) are used to initialize the values of a struct. 
+
+The `dot(.)` operator can also be used to assign values to the fields in the struct. In the example above, we create another variable called `s2` of type `Student` without initializing it. We then use the `dot(.)` operator to assign the value "Bharadwaj" to the name.
+
+### Putting it all together
+
+We are going to write a program to create a `struct` called "Student", get the details of 5 students, and display them to the user.
+
+```go
+package main
+
+import "fmt"
+
+type Student struct {
+    name string
+    age int
+    score float64
+}
+
+func get_details() {
+
+    var array[5] Student
+    fmt.Println("Enter the details of 5 students")
+
+    for i := 0; i < 5; i++ {
+        fmt.Println("Enter the name: ")
+        fmt.Scanln( & array[i].name)
+        fmt.Println("Enter the age: ")
+        fmt.Scanln( & array[i].age)
+        fmt.Println("Enter the score: ")
+        fmt.Scanln( & array[i].score)
+    }
+
+    fmt.Println("The details of the 5 students are as follows: ")
+    for i := 0; i < 5; i++ {
+        fmt.Println("Name:", array[i].name)
+        fmt.Println("Age: ", array[i].age)
+        fmt.Println("Score: ", array[i].score)
+    }
+}
+
+func main() {
+    get_details()
+}
+```
+
+The function `get_details()` gets the details of 5 students from the user and displays them on the terminal using a `for` loop. We create an array of type Student with length 5. This array stores the details of all 5 students. 
+
```

Comment:
A conclusion at the end of the article that quickly explains what this article covered would be a worthy addition 🥇